### PR TITLE
Patchify improvements

### PR DESCRIPTION
**Pull request by geokala**

### CODE REVIEW

File: patchify [hunk 1]
```diff
@@ -329,6 +333,75 @@ def ssh(manager_connection_string, keyfile, command):
     return subprocess.check_output(ssh_command)
 
 
+def _scp(keyfile, source, dest):
+    scp_command = ['scp']
+    if keyfile:
```
Comment (tehasdf) [inline patchify:392]: if `keyfile` is optional, it should likely be the last arg with a default None

File: patchify [hunk 2]
```diff
@@ -333,6 +336,23 @@ def ssh(manager_connection_string, keyfile, command):
     return subprocess.check_output(ssh_command)
 
 
+@contextmanager
+def temporarily_own_directory(manager_connection_string, keyfile, path):
```
Comment (tehasdf) [inline patchify:363]: no need for `temporarily` in the name - it's obvious that temporarily is why this is a context manager

File: patchify [hunk 3]
```diff
@@ -333,6 +336,23 @@ def ssh(manager_connection_string, keyfile, command):
...
+    ssh(
+        manager_connection_string,
+        keyfile,
+        'sudo chown $(whoami). ' + path,
```
Comment (tehasdf) [inline patchify:367]: dot `.` works here? or did you mean `:`

File: patchify [hunk 4]
```diff
@@ -837,9 +850,107 @@ def apply_patch(definition_file, manager_connection_string, keyfile,
...
+    current_patches.append(new_patch)
+
+    registry['current_patches'] = current_patches
+
+    tmp_path = tempfile.mkdtemp(prefix='patchify-')
+    local_reg_path = os.path.join(tmp_path, PATCH_REGISTRY_FILE_NAME)
+
+    try:
+        with open(local_reg_path, 'w') as local_reg_handle:
+            local_reg_handle.write(json.dumps(registry))
```
Comment (tehasdf) [inline patchify]: no `json.dump`? :P

**Review by tehasdf — APPROVED** [review]

File: patchify (hunk 3 above)
Comment (geokala) [reply patchify:367]: Dot does indeed work there- does the same as colon in this context (set group to default group of the user).

**Pull request merged**